File: src/EcommerceDDD.Spa/src/app/modules/ecommerce/models/requests/RegisterCustomerRequest.ts
export class RegisterCustomerRequest {
  email: string;
  name: string;
  shippingAddress: string;
  password: string;
  passwordConfirm: string;
  creditLimit: number;

  constructor(
    email: string,
    name: string,
    shippingAddress: string,
    password: string,
    passwordConfirm: string,
    creditLimit: number
  ) {
    this.email = email;
    this.name = name;
    this.shippingAddress = shippingAddress;
    this.password = password;
    this.passwordConfirm = passwordConfirm;
    this.creditLimit = creditLimit;
  }
}

File: src/Services/EcommerceDDD.Customers/API/Controllers/Requests/RegisterCustomerRequest.cs
﻿namespace EcommerceDDD.Customers.API.Controllers.Requests;

public record class RegisterCustomerRequest
{
    [Required(ErrorMessage = "The {0} field is required.")]
    [EmailAddress(ErrorMessage = "The field {0} is in an invalid format")]
    public string Email { get; init; }

    [Required(ErrorMessage = "The {0} field is required.")]
    [StringLength(100, ErrorMessage = "The {0} field must be between {2} and {1} characters", MinimumLength = 5)]
    public string Password { get; init; }

    [Compare("Password", ErrorMessage = "Passwords do not match.")]
    public string PasswordConfirm { get; init; }

    [Required(ErrorMessage = "The {0} field is required.")]
    [StringLength(100, ErrorMessage = "The {0} field must be between {2} and {1} characters", MinimumLength = 2)]
    public string Name { get; init; }

    [StringLength(100, ErrorMessage = "The {0} field must be between {2} and {1} characters", MinimumLength = 5)]
    public string ShippingAddress { get; init; }

    [Required(ErrorMessage = "The {0} field is required.")]
    public decimal CreditLimit { get; init; }
}